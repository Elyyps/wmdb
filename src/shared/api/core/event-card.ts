export interface IEventCard {
  buttonText: string;
  image: string;
  title: string;
}
