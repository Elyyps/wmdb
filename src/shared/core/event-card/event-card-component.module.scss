.event-card {
  &-head {
    margin-bottom: 32px;
    img {
      object-fit: cover;
      height: 148px;
      width: 308px;
      border-radius: 4px;
      @media (max-width: $breakpoint-xsmall-max) {
        width: 100%;
        height: 250px;
      }
    }
  }
  &-title {
    h4 {
      font-family: $global-font-family-secondary;
    }
  }
}
