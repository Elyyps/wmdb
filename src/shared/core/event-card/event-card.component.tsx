import * as React from "react";
import styles from "./event-card-component.module.scss";
import { LinkComponent } from "../link";

export interface IEventCardComponentProps {
  image?: string;
  link: string;
  title?: string;
}

const EventCardComponent = ({ title, image, link }: IEventCardComponentProps) => (
  <div className={styles["event-card"]}>
    <div className={styles["event-card-head"]}>
      <LinkComponent to={link}>
        <img src={image} alt="image" />
      </LinkComponent>
      <div className={styles["event-card-title"]}>
        <h4>{title}</h4>
      </div>
    </div>
  </div>
);

export { EventCardComponent };
