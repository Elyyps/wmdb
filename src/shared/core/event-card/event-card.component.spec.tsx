import React from "react";
import { shallow } from "enzyme";
import { EventCardComponent } from "./event-card.component";

describe("[EventCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<EventCardComponent link="" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
