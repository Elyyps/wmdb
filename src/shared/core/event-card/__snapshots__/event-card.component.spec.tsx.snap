// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EventCard] should render component without crashing 1`] = `
<div
  className="event-card"
>
  <div
    className="event-card-head"
  >
    <LinkComponent
      to=""
    >
      <img
        alt="image"
      />
    </LinkComponent>
    <div
      className="event-card-title"
    >
      <h4 />
    </div>
  </div>
</div>
`;
